spec/index.bs

```diff
@@ -6792,68 +6781,58 @@ typedef (sequence<double> or GPUColorDict) GPUColor;
 Note: `double` is large enough to precisely hold 32-bit signed/unsigned
 integers and single-precision floats.
 
-<script type=idl>
-dictionary GPUOrigin2DDict {
-    GPUIntegerCoordinate x = 0;
-    GPUIntegerCoordinate y = 0;
-};
-typedef (sequence<GPUIntegerCoordinate> or GPUOrigin2DDict) GPUOrigin2D;
-</script>
+<!--
+It's a bit of a hack to override the usual conversion to sequence.
+But if we did it explicitly (e.g. with a "typecheck" algorithm), it would infect
+every dictionary that contains one of these members, and every method that takes
+any of those dictionaries or this type.
+-->
 
-<script type=idl>
-dictionary GPUOrigin3DDict {
-    GPUIntegerCoordinate x = 0;
-    GPUIntegerCoordinate y = 0;
-    GPUIntegerCoordinate z = 0;
-};
-typedef (sequence<GPUIntegerCoordinate> or GPUOrigin3DDict) GPUOrigin3D;
-</script>
+## GPUOrigin2D ## {#gpuorigin2d}
```

Comment:
do we still need separate GPUOrigin2D and GPUOrigin3D types, now that everything is a sequence?

Reply:
They're more than just sequences; they're sequences of exactly 2 elements and 3 elements respectively. They would only be the same if we moved validation out of the types and into the methods.

Reply:
Wait, doesn't `GPUOrigin3D`, for example, a subject to the same confusion as the extent? I.e. in the dictionary it's not clear if `z` is for the depth or the array layer offset.

Reply:
Sort of yes, except calling array layer count "depth" is kind of a contradiction, whereas calling it [2] is neutral.